pkg/smartcontract/manifest/manifest.go

```diff
@@ -142,7 +144,24 @@ func (m *Manifest) ToStackItem() (stackitem.Item, error) {
 	}
 	extra := stackitem.Make("null")
 	if m.Extra != nil {
-		extra = stackitem.NewByteArray(m.Extra)
+		var err error
+		if !bytes.Equal([]byte("null"), m.Extra) {
+			// The result is put directly in the database and affects state-root calculation,
+			// thus use ordered map to stay compatible with C# implementation.
+			var mp ojson.OrderedObject
+			err = ojson.Unmarshal(m.Extra, &mp)
+			if err == nil {
+				var res []byte
+				res, err = ojson.Marshal(mp)
+				if err == nil { // Should always be nil, but who knows.
```

Comment:
Then maybe return an error if so?

Reply:
`json.RawMessage` should ensure validity automatically. But `Extra` field can in theory be an array.

Reply:
Well we must support this case as well, will check the behaviour of C# node.